### PR TITLE
degrade to armv8.0

### LINKED ISSUE #6254: 6.4.0 fails to start on Phytium arm platform
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Start tiflash 6.4.0 on specific arm platforms

### 2. What did you expect to see? (Required)
Started successfully like 6.3.0
image
### 3. What did you see instead (Required)
Compared with 6.3.0, there are some more link libraries：libgmssl.so    libgmssl.so.3.0 libgmssl.so.3 
image

### 4. What is your TiFlash version? (Required)
v6.4.0-pre

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
Signed-off-by: Lloyd-Pottiger <yan1579196623@gmail.com>

### What problem does this PR solve?

Issue Number: close #6254 

Problem Summary:

### What is changed and how it works?

* disable sve on ARM by default
* fix the broken test cases introduced by https://github.com/pingcap/tidb/pull/37384 and https://github.com/pingcap/tidb/pull/38447 that change the default value of `tidb_enable_clustered_index` from `INT_ONLY` to `ON`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: cmake/cpu_features.cmake [hunk 1]
```diff
@@ -36,7 +36,8 @@ if (ARCH_AARCH64)
     # [1] https://en.wikipedia.org/wiki/AArch64
     option (TIFLASH_ENABLE_ASIMD_SUPPORT "Enable Advanced SIMD support." ON)
     option (TIFLASH_ENABLE_SVE_SUPPORT "Enable Scalable Vector Extension support." OFF)
```
Comment (solotzg) [inline cmake/cpu_features.cmake:38]: `TIFLASH_ENABLE_ASIMD_SUPPORT` is used as default feature in v6.3.

Comment (Lloyd-Pottiger) [reply cmake/cpu_features.cmake:38]: fixed

**Review by solotzg — APPROVED** [review]
lgtm

File: cmake/cpu_features.cmake [hunk 2]
```diff
@@ -36,12 +36,18 @@ if (ARCH_AARCH64)
...
-    option (NO_ARMV81_OR_HIGHER "Disable ARMv8.1 or higher on Aarch64 for maximum compatibility with older/embedded hardware." OFF)
+    # TODO: default ON, to be changed after CI is updated
+    option (NO_ARMV81_OR_HIGHER "Disable ARMv8.1 or higher on Aarch64 for maximum compatibility with older/embedded hardware." ON)
 
     if (NO_ARMV81_OR_HIGHER)
         # crc32 is optional in v8.0 and mandatory in v8.1. Enable it as __crc32()* is used in lot's of places and even very old ARM CPUs
         # support it.
-        set (COMPILER_FLAGS "${COMPILER_FLAGS} -march=armv8+crc")
+        set (TEST_FLAG "-march=armv8+crc")
```
Comment (solotzg) [inline cmake/cpu_features.cmake:45]: Are these `TEST_FLAG` used anywhere?

Comment (Lloyd-Pottiger) [reply cmake/cpu_features.cmake:45]: in line 50, `set (COMPILER_FLAGS "${COMPILER_FLAGS} ${TEST_FLAG}")` append it to COMPILER_FLAGS.

Comment (solotzg) [reply cmake/cpu_features.cmake:45]: If there is no need to invoke `check_cxx_source_compiles`, why not set `COMPILER_FLAGS` directly.

Comment (Lloyd-Pottiger) [reply cmake/cpu_features.cmake]: fixed

File: tests/fullstack-test2/ddl/alter_pk.test [hunk 3]
```diff
@@ -62,7 +62,7 @@ mysql> alter table test.t drop primary key;
 mysql> drop table if exists test.t_case;
 ## create table with `source` is nullable
 ## insert some data and left `source` to be empty
-mysql> create table test.t_case (`case_no` varchar(32) not null,`source` varchar(20) default null,`p` varchar(12) DEFAULT NULL,primary key (`case_no`));
+mysql> create table test.t_case (`case_no` varchar(32) not null,`source` varchar(20) default null,`p` varchar(12) DEFAULT NULL,primary key (`case_no`) NONCLUSTERED);
```
Comment (JaySon-Huang) [inline tests/fullstack-test2/ddl/alter_pk.test:65]: Introduced by https://github.com/pingcap/tidb/pull/37384 and https://github.com/pingcap/tidb/pull/38447

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**